scripts/scil_split_volume_by_ids.py

```diff
@@ -89,31 +63,25 @@ def main():
     for label, name in zip(label_indices, label_names):
         if int(label) != 0:
             if args.out_prefix:
-                output_filenames.append(os.path.join(args.out_dir,
-                                                     '{0}_{1}.nii.gz'.format(
-                                                         args.out_prefix,
-                                                         name)))
+                output_filenames.append(os.path.join(
+                    args.out_dir,
+                    '{0}_{1}.nii.gz'.format(args.out_prefix, name)))
             else:
-                output_filenames.append(os.path.join(args.out_dir,
-                                                     '{0}.nii.gz'.format(
-                                                         name)))
+                output_filenames.append(os.path.join(
+                    args.out_dir, '{0}.nii.gz'.format(name)))
 
     assert_output_dirs_exist_and_empty(parser, args, [], optional=args.out_dir)
     assert_outputs_exist(parser, args, output_filenames)
 
     # Extract the voxels that match the label and save them to a file.
-    cnt_filename = 0
-    for label in label_indices:
-        if int(label) != 0:
-            split_label = np.zeros(label_img.shape,
-                                   dtype=np.uint16)
-            split_label[np.where(label_img_data == int(label))] = label
+    split_data = split_labels(label_img_data, label_indices)
 
-            split_image = nib.Nifti1Image(split_label,
+    for i in range(len(label_indices)):
```

Comment:
This list is always out of range singe label_indices include the 0 index, but the filenames excludes it.

Right now labels 2 is saved as PREFIX_1.nii.gz (shifted)

Reply:
I'm not sure that this was the issue. I think I fixed it nonetheless. Can you verify?

Reply:
Fixed.